### PR TITLE
Cleaned up renderers cache and added unit tests

### PR DESCRIPTION
**Pull request by imalberto**

### CODE REVIEW

File: lib/app/addons/ac/output-adapter.common.js [hunk 1]
```diff
@@ -201,23 +200,24 @@ YUI.add('mojito-output-adapter-addon', function(Y, NAME) {
...
-            if (rendererCache.hb === undefined || rendererCache.mu === undefined) {
-                rendererCache = {
-                    hb: new (Y.mojito.addons.viewEngines.hb)(''), // viewId
-                    mu: new (Y.mojito.addons.viewEngines.mu)('', {}) // viewId, viewOptions
-                };
+            // optimize for server only
+            if (typeof context.runtime !== undefined && 'server' === context.runtime) {
+                renderer = CACHE.renderers[mojitView.engine];
+                if (!renderer) {
+                    CACHE.renderers[mojitView.engine] = renderer =
```
Comment (drewfish) [inline lib/app/addons/ac/output-adapter.common.js:209]: Does jslint complain about this double assignment?

File: lib/app/addons/ac/output-adapter.common.js [hunk 2]
```diff
@@ -201,23 +200,24 @@ YUI.add('mojito-output-adapter-addon', function(Y, NAME) {
...
+                        new (Y.mojito.addons.viewEngines[mojitView.engine])('', {});
```
Comment (drewfish) [inline lib/app/addons/ac/output-adapter.common.js]: It looks like, on the server, that Y.mojito.ViewRenderer isn't being used. That's slightly inconsistent, but it looks like it'll be OK.

**Comment by drewfish** [comment]
It looks like the view engines are no longer getting `viewEngineOptions` when run on the server.

File: lib/app/autoload/dispatch.common.js [hunk 3]
```diff
@@ -5,7 +5,7 @@
  */
 
 /*jslint anon:true, sloppy:true, nomen:true*/
-/*global YUI*/
+/*global YUI,window*/
```
Comment (drewfish) [inline lib/app/autoload/dispatch.common.js]: This seems kinda random.  What does app/autoload/dispatch.common.js have to do with view engines?

**Comment by imalberto** [comment]
@drewfish
- nice catch about the viewEngineOptions
- mojito jslint did not complain about the double assignment
- undo the change to dispatch.common. nothing to do with output adapter.

**Comment by drewfish** [comment]
Could you perhaps add a test, to make sure that `viewEngineOptions` doesn't get lost again in the future?

**Comment by drewfish** [comment]
+1

**Pull request merged**